lib/widgets/totals_table.dart

```diff
@@ -1,79 +1,92 @@
 import 'package:flutter/material.dart';
+import 'package:rationapp/models/cow_characteristics_model.dart';
 import '../models/cow_requirements_model.dart';
 
 class TotalsTable extends StatelessWidget {
   final List<Map<String, dynamic>> fodderItems;
   final List<Map<String, dynamic>> concentrateItems;
   final CowRequirements cowRequirements;
+  final CowCharacteristics cowCharacteristics;
 
-  TotalsTable({
-    super.key,
-    required this.fodderItems,
-    required this.concentrateItems,
-    required this.cowRequirements,
-  });
-
-  final List<Map<String, dynamic>> columns = const [
-    {
-      'label': 'DM Intake\n(kg/d)',
-      'key': 'dmIntake',
-      'decimals': 2,
-      'reqKey': 'dmIntake',
-      'isPercentage': false,
-    },
-    {
-      'label': 'ME Intake\n(MJ/d)',
-      'key': 'meIntake',
-      'decimals': 2,
-      'reqKey': 'meIntake',
-      'isPercentage': false,
-    },
-    {
-      'label': 'CP Intake\n(%)',
-      'key': 'cpIntake',
-      'decimals': 2,
-      'reqKey': 'cpIntake',
-      'isPercentage': true,
-    },
-    {
-      'label': 'NDF Intake\n(%)',
-      'key': 'ndfIntake',
-      'decimals': 2,
-      'reqKey': 'ndfIntake',
-      'isPercentage': true,
-    },
-    {
-      'label': 'Ca Intake\n(%)',
-      'key': 'caIntake',
-      'decimals': 2,
-      'reqKey': 'caIntake',
-      'isPercentage': true,
-    },
-    {
-      'label': 'P Intake\n(%)',
-      'key': 'pIntake',
-      'decimals': 2,
-      'reqKey': 'pIntake',
-      'isPercentage': true,
-    },
-    {
-      'label': 'Concentrate Intake\n(%)',
-      'key': 'concentrateIntake',
-      'decimals': 2,
-      'reqKey': 'concentrateIntake',
-      'isPercentage': true,
-    },
-    {
-      'label': 'Cost\n(ERN)',
-      'key': 'cost',
-      'decimals': 2,
-      'reqKey': null,
-      'isPercentage': false,
-    },
-  ];
+  TotalsTable(
+      {super.key,
+      required this.fodderItems,
+      required this.concentrateItems,
+      required this.cowRequirements,
+      required this.cowCharacteristics});
+
+  bool get isLactating => cowCharacteristics.lactationStage != 'Dry';
+
+  List<Map<String, dynamic>> get columns {
+    return [
+      {
+        'label': 'DM Intake\n(kg/d)',
+        'key': 'dmIntake',
+        'decimals': 2,
+        'reqKey': 'dmIntake',
+        'min': 0.95,
+        'max': 1.05
+      },
+      {
+        'label': 'ME Intake\n(MJ/d)',
+        'key': 'meIntake',
+        'decimals': 2,
+        'reqKey': 'meIntake',
+        'min': 0.95,
```

Comment:
This is multiplied by the specific cow's requirements, and is not an absolute value